falcon/api.py

```diff
@@ -347,7 +352,14 @@ def add_route(self, uri_template, resource, *args, **kwargs):
         if '//' in uri_template:
             raise ValueError("uri_template may not contain '//'")
 
-        method_map = routing.map_http_methods(resource)
+        # NOTE(santeyio): This is a not very nice way to catch the suffix
+        # keyword. In python 3 it can be specified explicitly in the function
+        # definition, e.g.
+        # `add_route(self, uri_template, resource, *args, suffix=None, **kwargs)`
```

Comment:
That is why I was proposing a separate method for it, something like

``add_alternate_route(self, uri_template, suffix, *args, **kwargs)``

This would also eliminate the risk that someone already had a ``suffix`` inside ``kwargs`` for some other purpose. On the other hand, 2.0 is about being able to have breaking changes, and having the same ``add_route`` method probably offers simplicity and brevity over 2 methods.

Reply:
Yeah, I was kind of thinking it may be a bit more elegant to just add the kwarg. As you say, in the off chance it breaks someone's custom router it is a major release after all. I don't want to just go breaking things left and right, but this seems like a low-risk change.